src/Discord.Net.SlashCommands/TypeConverters/DefaultValueConverter.cs

```diff
@@ -0,0 +1,61 @@
+using Discord.WebSocket;
+using System;
+using System.Threading.Tasks;
+
+namespace Discord.SlashCommands
+{
+    internal class DefaultValueConverter<T> : TypeConverter<T> where T : IConvertible
+    {
+        public override ApplicationCommandOptionType GetDiscordType ( )
+        {
+            switch (Type.GetTypeCode(typeof(T)))
+            {
+                case TypeCode.Boolean:
+                    return ApplicationCommandOptionType.Boolean;
+
+                case TypeCode.DateTime:
+                case TypeCode.SByte:
+                case TypeCode.Byte:
+                case TypeCode.Char:
+                case TypeCode.String:
+                case TypeCode.Single:
+                    return ApplicationCommandOptionType.String;
+
+                case TypeCode.Int16:
+                case TypeCode.Int32:
+                case TypeCode.Int64:
+                case TypeCode.UInt16:
+                case TypeCode.UInt32:
+                case TypeCode.UInt64:
+                    return ApplicationCommandOptionType.Integer;
+
+                case TypeCode.Decimal:
+                case TypeCode.Double:
+                    return ApplicationCommandOptionType.Number;
+
+                case TypeCode.DBNull:
+                default:
+                    throw new InvalidOperationException($"Parameter Type {typeof(T).FullName} is not supported by Discord.");
+            }
+        }
+        public override Task<TypeConverterResult> ReadAsync (ISlashCommandContext context, SocketSlashCommandDataOption option, IServiceProvider services)
```

Comment:
Consider using a value task

Reply:
User defined TypeConverters should be able to pass additional info to the InteractionService in case of failure